x-pack/plugins/encrypted_saved_objects/server/create_migration.test.ts

```diff
@@ -163,6 +165,80 @@ describe('createMigration()', () => {
         attributes
       );
     });
+
+    describe('uses the object `namespaces` field to populate the descriptor when the migration context indicates this type is being converted', () => {
+      const doTest = async ({
```

Comment:
I am... not sure why but when I make this change I get jest test failures:

```
  ● createMigration() › migration of a single legacy type › uses the object `namespaces` field to populate the descriptor when the migration context indicates this type is being converted › when the first namespace element is another string

    expect(jest.fn()).toHaveBeenCalledWith(...expected)

    Expected: {"id": "123", "namespace": "foo", "type": "known-type-1"}, {"firstAttr": "first_attr"}, {"convertToMultiNamespaceType": true}

    Number of calls: 0

      209 |         );
      210 | 
    > 211 |         expect(serviceWithLegacyType.decryptAttributesSync).toHaveBeenCalledWith(
          |                                                             ^
      212 |           {
      213 |             id: '123',
      214 |             type: 'known-type-1',

      at doTest (x-pack/plugins/encrypted_saved_objects/server/create_migration.test.ts:211:61)
      at Object.<anonymous> (x-pack/plugins/encrypted_saved_objects/server/create_migration.test.ts:239:9)
```

Reply:
> I am... not sure why but when I make this change I get jest test failures:

And that's the power of tests, they fail when they are wrong! :rofl: But jokes aside, the tests with `async` "work" for the same reason why the test below works too - without `await` test doesn't wait until promise is resovled and `expect`s inside of a promise are not properly handled (they [are logged though](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105821/execution/node/194/log/?consoleFull), starting at `23:19:17`):

```ts
it('WAAAT?', () => {
  new Promise((resolve) => {
    expect(false).toBe(true);
    resolve(undefined);
  });
});
```

Having said that I know that was a problem in the past, but I had an impression that since some time ago our Jest tests started to fail because of unhandled rejected promises. @elastic/kibana-operations am I making this up?

Reply:
I'll chalk this up to a brain-fart moment :) Fixed in 9d77ce86